### PR TITLE
Multiple enhancements, adjustments, and bug fixes to frontend code

### PR DESCRIPTION
**Pull request by rubenthoms**
- Improved styling multiple places
- Added `MenuButton` component
- Improved color palette "Red to blue"
- Bug fix in `Input` component
- Added serialization method to `ColorPalette` and `ColorScale`
- Adjusted wellbore selector in 3D viewer
- Made `SubsurfaceViewerWithCameraState` a shared component
- Removed wrongly located `package-lock.json`

### CODE REVIEW

File: frontend/src/lib/components/Input/input.tsx [hunk 1]
```diff
@@ -106,6 +106,22 @@ export const Input = React.forwardRef((props: InputProps, ref: React.ForwardedRe
...
+                        newValue = Math.max(props.min, newValue);
+                    }
+
+                    if (props.max !== undefined) {
+                        newValue = Math.min(props.max, newValue);
+                    }
+                }
+
+                event.target.value = newValue.toString();
+            }
```
Comment (jorgenherje) [inline frontend/src/lib/components/Input/input.tsx:124]: Should the `setValue` on line 106 be triggered after adjustment/configuration of `event.target.value`, i.e. move the if-check and value adjustment outside of the `props.onChange`-if and before the `setValue`?

File: frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx [hunk 2]
```diff
@@ -12,49 +15,122 @@ export type WellboreSelectorProps = {
...
+    const [selectedWellborePurposes, setSelectedWellborePurposes] = useValidArrayState<string>({
+        initialState: availableWellborePurposes,
+        validStateArray: availableWellborePurposes,
+    });
+    React.useEffect(() => {
+        onSelectedWellboreUuidsChange(
+            props.wellboreHeaders
+                .filter((header) => selectedWellborePurposes.includes(header.wellborePurpose))
+                .filter((header) => selectedWellboreStatuses.includes(header.wellboreStatus))
+                // .filter((header) => props.selectedWellboreUuids.includes(header.wellboreUuid))
```
Comment (jorgenherje) [inline frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx:36]: Can the two sequential `filter` be refactored into a reusable function, as it seems to be reused multiple times in the component. E.g. something providing an array `selectedWellBoreHeader`

... I'm assuming we can remove the third unused/commented `.filter` function here?

File: frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx [hunk 3]
```diff
@@ -12,49 +15,122 @@ export type WellboreSelectorProps = {
...
+                .map((header) => header.wellboreUuid)
+        );
+    }, [selectedWellborePurposes, selectedWellboreStatuses, onSelectedWellboreUuidsChange, props.wellboreHeaders]);
+
     function handleSelectAll() {
-        props.onSelectedWellboreUuidsChange(props.wellboreHeaders.map((header) => header.wellboreUuid));
+        props.onSelectedWellboreUuidsChange(
+            props.wellboreHeaders
+                .filter((header) => selectedWellborePurposes.includes(header.wellborePurpose))
+                .filter((header) => selectedWellboreStatuses.includes(header.wellboreStatus))
```
Comment (jorgenherje) [inline frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx:45]: Consider: Make reusable `filter` function

File: frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx [hunk 4]
```diff
@@ -12,49 +15,122 @@ export type WellboreSelectorProps = {
...
+                .map((header) => header.wellboreUuid)
+        );
     }
     function handleUnselectAll() {
         props.onSelectedWellboreUuidsChange([]);
     }
+    function makeWellHeadersOptions(): SelectOption[] {
+        return props.wellboreHeaders
+            .filter((header) => selectedWellboreStatuses.includes(header.wellboreStatus))
+            .filter((header) => selectedWellborePurposes.includes(header.wellborePurpose))
```
Comment (jorgenherje) [inline frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx:55]: Consider: Make reusable `filter` function

File: frontend/src/modules/_shared/components/ColorLegendsContainer/colorLegendsContainer.tsx [hunk 5]
```diff
@@ -4,23 +4,26 @@ import { ColorScale, ColorScaleGradientType, ColorScaleType } from "@lib/utils/C
...
-const lineColor = "#555";
-const textGap = 6;
-const offset = 10;
-const legendGap = 4;
-const textWidth = 70;
-const nameLabelWidth = 10;
-const fontSize = 10;
-
-const textStyle: React.CSSProperties = {
+const PREFERENCES = {
```
Comment (jorgenherje) [inline frontend/src/modules/_shared/components/ColorLegendsContainer/colorLegendsContainer.tsx:7]: Is "Internal Style" or something similar a better name? As `PREFERENCES` is not really a preference/opt-in, but actually a style added in addition to the `ColorLegendProp` provided to the component

**Review by jorgenherje — CHANGES_REQUESTED** [review]
Nice adjustments, some minor comments

File: frontend/src/framework/components/EnsembleDropdown/ensembleDropdown.tsx [hunk 6]
```diff
@@ -1,24 +1,29 @@
+import { Ensemble } from "@framework/Ensemble";
 import { EnsembleIdent } from "@framework/EnsembleIdent";
 import { EnsembleSet } from "@framework/EnsembleSet";
 import { ColorTile } from "@lib/components/ColorTile";
 import { Dropdown, DropdownOption, DropdownProps } from "@lib/components/Dropdown";
 
 type EnsembleDropdownProps = {
-    ensembleSet: EnsembleSet;
+    ensembleSet?: EnsembleSet;
+    ensembles?: Ensemble[];
```
[the remark this answers is not in the document]
Comment (rubenthoms) [reply frontend/src/framework/components/EnsembleDropdown/ensembleDropdown.tsx]: Agreed and added change to PR.

File: frontend/src/lib/components/Input/input.tsx (hunk 1 above)
Comment (rubenthoms) [reply frontend/src/lib/components/Input/input.tsx:124]: As discussed yesterday, they should not be synced in order to allow intermediate invalid states locally without publishing them to parent.

File: frontend/src/framework/WorkbenchSession.ts [hunk 7]
```diff
@@ -77,7 +77,7 @@ export class WorkbenchSession {
     }
 }
 
-function createEnsembleRealizationFilterFuncForWorkbenchSession(workbenchSession: WorkbenchSession) {
+export function createEnsembleRealizationFilterFuncForWorkbenchSession(workbenchSession: WorkbenchSession) {
```
[the remark this answers is not in the document]
Comment (rubenthoms) [reply frontend/src/framework/WorkbenchSession.ts:80]: As discussed yesterday, this function is used in a class implementation which cannot consume a hook.

File: frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx (hunk 2 above)
Comment (rubenthoms) [reply frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx:36]: Sure. This whole file is an intermediate state and requires some refactoring anyways. I created a common util function for now.

File: frontend/src/modules/_shared/components/ColorLegendsContainer/colorLegendsContainer.tsx (hunk 5 above)
Comment (rubenthoms) [reply frontend/src/modules/_shared/components/ColorLegendsContainer/colorLegendsContainer.tsx:7]: Going with "STYLE_CONSTANTS".

**Review by jorgenherje — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**